app/scripts/lib/metrics.js

```diff
@@ -153,6 +159,66 @@ define(function (require, exports, module) {
       this._clearInactivityFlushTimeout();
     },
 
+    notifications: {
+      /* eslint-disable sorting/sort-object-props */
+      'flow.initialize': '_initializeFlowModel',
+      'flow.event': '_logFlowEvent'
+      /* eslint-enable sorting/sort-object-props */
+    },
+
+    /**
+     * Initialize the flow model. If it's already been initalized, do nothing.
+     * Initialization may fail if the required flow properties can't be found,
+     * either in the DOM or the resume token.
+     */
+    _initializeFlowModel () {
+      if (this._flowModel) {
+        return;
+      }
+
+      const flowModel = new Flow({
+        sentryMetrics: this._sentryMetrics,
```

Comment:
Not for this PR - I wonder if we can remove the `_sentryMetrics` reference from the Flow model and pass it a notifier instead, which would also mean the metrics model would no longer need a reference to `_sentryMetrics`. Whenever the Flow model needs to record an error, it could trigger a message on the notifier that makes its way to both the normal metrics module and Sentry. Lots of ✋ 👋 (hand-waving) here, but it seems feasible.